ext/mbstring/mbstring.c

```diff
@@ -1279,18 +1280,15 @@ PHP_FUNCTION(mb_http_input)
 			if (n == 0) {
 				RETURN_FALSE;
 			}
-			// TODO Use smart_str instead.
-			mbfl_string result;
-			mbfl_memory_device device;
-			mbfl_memory_device_init(&device, n * 12, 0);
+
+			smart_str result = {0};
 			for (size_t i = 0; i < n; i++, entry++) {
-				mbfl_memory_device_strcat(&device, (*entry)->name);
-				mbfl_memory_device_output(',', &device);
+				if (i > 0) {
+					smart_str_appendc(&result, ',');
+				}
+				smart_str_appends(&result, (*entry)->name);
 			}
-			mbfl_memory_device_unput(&device); /* Remove trailing comma */
-			mbfl_memory_device_result(&device, &result);
-			RETVAL_STRINGL((const char*)result.val, result.len);
-			mbfl_string_clear(&result);
+			RETVAL_STR(smart_str_extract(&result));
 			return;
```

Comment:
Nit:
```suggestion
			RETURN_STR(smart_str_extract(&result));
```

Reply:
Nice, thanks. Adjusted.

Reply:
Did you forget to push the commit?

Reply:
Sorry, I was intending to do it but delayed. Pushed now.